serokell/tezos-packaging
# [#349] Add custom chains baking services

## Description

This PR adds a convenient way to start a baking service on a network with a custom chain which is defined by the custom node config file.

Testing changes locally:

1) Install updated `tezos-node` and `tezos-baking packages`
2) Clone [`local-chain` repo](https://gitlab.com/morley-framework/local-chain).
3) Create a config file for future custom baking instance:
```
sudo cp /etc/default/tezos-baking-custom@ /etc/default/tezos-baking-custom@voting
```
4) Start `./scenarios/voting.py --base-protocol hangzhou` from `local-chain`.
5) Fill `CUSTOM_NODE_CONFIG` in `/etc/default/tezos-baking-custom@voting` with the path provided by the voting scenario.
6) Start custom baking service:
```
sudo systemctl start tezos-baking-custom@voting
```
7) Check that all parts of it are actually running:
```
systemctl status tezos-node-custom@voting
systemctl status tezos-baker-011-pthangz2@custom@voting.service
systemctl status tezos-baker-012-psithaca@custom@voting.service
systemctl status tezos-endorser-011-pthangz2@custom@voting.service
```
8) Follow the voting scenario using the node from the `tezos-node-custom@voting`, it has `http://localhost:8732` endpoint by default.

9.1 Reset this custom baking service once the scenario is done:
```
sudo systemctl start tezos-baking-custom-reset@voting
```
9.2 Alternative way to reset custom baking service is to set `RESET_ON_STOP=true` env variable in config file and run
```
sudo systemctl stop tezos-baking-custom@voting
```

The only manual steps left are 4) and 5). I think they can be added to the `tezos-setup-wizard`.
Alternatively, we can consider custom baking services as an _advanced_ feature, leave these steps manual, and
document how one may start a custom baking service without the setup wizard. What do you think?



## Related issue(s)



Relates #349

#### Related changes (conditional)

- [ ] I checked whether I should update the [README](/serokell/tezos-packaging/tree/master/README.md)

- [x] I checked whether native packaging works, i.e. native binary packages
  can be successfully built.

#### Stylistic guide (mandatory)

- [x] My commits comply with [the policy used in Serokell](https://www.notion.so/serokell/Where-and-how-to-commit-your-work-58f8973a4b3142c8abbd2e6fd5b3a08e).
